docs/configuration_files.rst

```diff
@@ -7,92 +7,92 @@ Jwst_reffiles makes use of multiple configuration files to control the behavior
 
 .. _main_cfg:
 
-An example of the main configuraiton file:
+An example of the main configuration file:
 
-::
+.. parsed-literal::
 
-    instrument: NIRCam
-    reflabels: [example_readnoise_module]
-    reftypes: [bpm,rdnoise,gain,linearity]
+    instrument_: NIRCam
+    reflabels_: [example_readnoise_module]
+    reftypes_: [bpm,rdnoise,gain,linearity]
 
-    default_reflabels:
+    default_reflabels_:
        bpm: bpm
        rdnoise: example_readnoise_module
        gain: gain_armin
 
     inputfiles:
-       dark_pattern: Bias|Dark|BIAS|DARK|ZERO
-       flat_pattern: GAIN|LIN
+       dark_pattern_: Bias|Dark|BIAS|DARK|ZERO
+       flat_pattern_: GAIN|LIN
 
-       basenamepattern: (.*)\_uncal\.fits$
+       basenamepattern_: (.*)\_uncal\.fits$
 
        # specificy the fits keywords to determine MJD time of
        # observation. They are automatically added to the list of required
        # fits keywords. if MJD-OBS does not exist, then DATE-OBS (and
        # TIME-OBS if specified) are used to determine the MJD.
-       dateobs_fitskey: DATE-END
-       timeobs_fitskey: TIME-END
-       mjdobs_fitskey:
+       dateobs_fitskey_: DATE-END
+       timeobs_fitskey_: TIME-END
+       mjdobs_fitskey_:
 
-       requiredfitskeys: [INSTRUME,DETECTOR,NGROUPS,NINTS,TGROUP,SUBARRAY]
-       optionalfitskeys: [FILTER,PUPIL]
+       requiredfitskeys_: [INSTRUME,DETECTOR,NGROUPS,NINTS,TGROUP,SUBARRAY]
+       optionalfitskeys_: [FILTER,PUPIL]
 
-    DD:
-       max_Delta_MJD: 2.0
+    DD_:
+       max_Delta_MJD_: 2.0
 
-    FF:
-       max_Delta_MJD: 2.0
-    DDFF:
-       max_Delta_MJD: 10.0
+    FF_:
+       max_Delta_MJD_: 2.0
+    DDFF_:
+       max_Delta_MJD_: 10.0
 
-    # reference file output basename: outrootdir[/outsubdir][/runID]/reflabel/reflabel[_outsubdir][_runID][.addsuffix].cmdID.reftype.fits
+    # reference file output basename_: outrootdir[/outsubdir][/runID]/reflabel/reflabel[_outsubdir][_runID][.addsuffix].cmdID.reftype.fits
     # if runID=AUTO, then the latest runID in the output dir outrootdir[/outsubdir] is taken
     output:
-       outrootdir: $JWST_MKREFS_OUTDIR
-       outsubdir: test1
-       runID: AUTO
+       outrootdir_: $JWST_MKREFS_OUTDIR
```

Comment:
might want to specify you have to set this environment variable in the setup/installation section. If it's not set, does the code crash or does it just write to the current working directory?

Reply:
I'm not sure I've tried without having it defined. I'm sure it'll crash though when it tries to expand an environment variable that doesn't exist. We should make sure mkrefs.py checks that the output directory is a valid path. I've opened #46 to monitor this.